engine/src/parser/expression_utils.cpp

```diff
@@ -1496,3 +1496,15 @@ std::string preprocess_expression_for_evaluation(std::string expression, blazing
 	expression = apply_interval_conversion(expression, schema);
   return expression;
 }
+
+// This function exists because there was an error performing e2e tests about concat.
+// The function `convert_concat_expression_into_multiple_binary_concat_ops` was removed.
+std::string preprocess_expression_for_filter(std::string expression, blazingdb::manager::Context * context, std::vector<cudf::data_type> schema) {
+	expression = fill_minus_op_with_zero(expression);
+	// expression = convert_concat_expression_into_multiple_binary_concat_ops(expression);
```

Comment:
Do you know why `convert_concat_expression_into_multiple_binary_concat_ops` causes an error for e2e tests? Is there a particular query that is problematic? What happens to the expression here that makes it be a problem? Could it be that the issue is more that `convert_concat_expression_into_multiple_binary_concat_ops` does not properly handle some edge cases?

Reply:
Indeed, I updated this branch with the last fixes for concat-expressions. I commented `convert_concat_expression_into_multiple_binary_concat_ops` to indentify the issue source. Now I'm adapting it to cover the edge-cases.